api/v1alpha2/linodemachine_types.go

```diff
@@ -161,6 +161,11 @@ type LinodeMachineStatus struct {
 	// Addresses contains the Linode instance associated addresses.
 	Addresses []clusterv1.MachineAddress `json:"addresses,omitempty"`
 
+	// CloudinitMetadataSupport determines whether to use cloud-init or not.
+	// +optional
+	// +kubebuilder:default=true
+	CloudinitMetadataSupport bool `json:"cloudinitMetadataSupport,omitempty"`
```

Comment:
This seems to be a chance separate from the rest of the PR, do you think we now need this field as well?

Reply:
This is required as we store the result of getRegion() and getImage() in this variable. We set it in reconcilePreflightMetadataSupportConfigure(). It gets used later where we decide to use cloud-init or stackscripts based on the value of this variable.

Reply:
It will automatically set it. We are doing this in code here: **https://github.com/linode/cluster-api-provider-linode/pull/522/files#diff-8daab1045cecf480419b0a31c2f8558e51197d31a7be910386ef4e420db0f7a4R310**